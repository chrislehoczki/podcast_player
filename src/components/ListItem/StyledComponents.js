import styled from "styled-components";
import theme from "../../styles/theme";

export const ListContainer = styled.div`
  display: flex;
  padding: 10px;
  border: 1px solid ${theme.primary};
`;

export const DetailContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 10px;
  width: 400px;
`;

export const MainDetail = styled.h4`
  margin: 0;
`;

export const SubDetail = styled.div`
  margin: 0;
  font-size: 14px;
`;

export const Duration = styled.p`
  margin: 3px 0;
  font-size: 14px;
`;
