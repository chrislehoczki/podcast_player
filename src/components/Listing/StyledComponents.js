import styled from "styled-components";
import theme from "../../styles/theme";

export const Container = styled.div`
  width: 100%;
  height: calc(100vh - 40px - 400px);
  overflow-y: auto;
  ::-webkit-scrollbar {
    width: 10px;
  }
  ::-webkit-scrollbar-track {
    background: 0;
  }
  ::-webkit-scrollbar-thumb {
    background: ${theme.primary};
  }
  @media only screen and (max-width: 400px) {
    height: calc(100vh - 400px);
  }
`;
