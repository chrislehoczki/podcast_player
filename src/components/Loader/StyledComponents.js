import styled from "styled-components";

export const Ring = styled.div`
  display: inline-block;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  &:after {
    content: " ";
    display: block;
    width: 55%;
    height: 55%;
    margin: 8px;
    border-radius: 50%;
    border: 6px solid #fff;
    border-color: #fff transparent #fff transparent;
    animation: lds-dual-ring 1.2s linear infinite;
  }
  @keyframes lds-dual-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

export const LoaderContainer = styled.div`
  width: ${props => props.size}px;
  height: ${props => props.size}px;
  min-width: ${props => props.size}px;
  min-height: ${props => props.size}px;
  border: ${props => (props.border ? "2px solid black" : 0)};
  outline: 0;
  margin: 20px;
`;
