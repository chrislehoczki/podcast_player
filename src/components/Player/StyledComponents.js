import styled from "styled-components";
import theme from "../../styles/theme";

export const PlayerContainer = styled.div`
  width: 100%;
  height: 400px;
  position: relative;
  background: ${theme.background};
  border-bottom: 2px solid ${theme.primary};
`;

export const Image = styled.div`
  width: 150px;
  height: 150px;
  min-height: 150px;
  background: url(${props => props.img}) no-repeat center center;
  background-size: cover;
`;

export const Details = styled.div`
  height: 60%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  text-align: center;
  padding: 20px 10px;
`;

export const Title = styled.h1`
  margin: 0;
  font-size: 18px;
  margin: 20px 0 5px 0;
  line-height: 22px;
`;

export const Author = styled.h2`
  font-size: 16px;
  margin: 0;
`;

export const Controls = styled.div`
  display: flex;
  vertical-align: center;
  align-items: center;
  width: 100%;
  height: 20%;
  justify-content: center;
`;
