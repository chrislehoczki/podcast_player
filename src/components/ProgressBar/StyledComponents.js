import styled from "styled-components";
import theme from "../../styles/theme";

const sliderThumbStyles = props => `
  width: 15px;
  height: 15px;
  border-radius: 100%;
  background: ${theme.primary};
`;

export const Container = styled.div`
  position: absolute;
  bottom: 25px;
  width: 80%;
  margin: 0;
  display: flex;
  align-items: center;
  color: ${theme.primary};
  .slider {
    flex: 6;
    -webkit-appearance: none;
    width: 100%;
    height: 5px;
    border-radius: 5px;
    background: ${theme.primary};
    outline: none;
    &::-webkit-slider-thumb {
      -webkit-appearance: none;
      appearance: none;
      ${props => sliderThumbStyles(props)}
    }
    &::-moz-range-thumb {
      ${props => sliderThumbStyles(props)}
    }
  }
`;
