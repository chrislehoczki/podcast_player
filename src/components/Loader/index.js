import React from "react";
import propTypes from "prop-types";

import { Ring, LoaderContainer } from "./StyledComponents";
const Loader = ({ size }) => {
  return (
    <LoaderContainer size={size}>
      <Ring />
    </LoaderContainer>
  );
};

Loader.propTypes = {
  size: propTypes.number
};

export default Loader;
