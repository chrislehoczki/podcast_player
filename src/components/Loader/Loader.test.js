import React from "react";
import { shallow, mount } from "enzyme";
import Loader from "./index";
import { LoaderContainer, Ring } from "./StyledComponents";

describe("<Loader />", () => {
  it("Renders correctly", () => {
    const component = shallow(<Loader size={60} />);
    expect(component).toMatchSnapshot();
  });
  it("Renders a container of correct size", () => {
    const component = mount(<Loader size={60} />);
    expect(component.find(LoaderContainer)).toHaveStyleRule("width", "60px");
  });
  it("Renders a loader ring", () => {
    const component = shallow(<Loader size={60} />);
    expect(component.find(LoaderContainer)).toExist();
  });
});
