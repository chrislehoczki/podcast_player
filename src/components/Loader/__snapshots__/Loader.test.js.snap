// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Renders correctly 1`] = `
<styled.div
  size={60}
>
  <styled.div />
</styled.div>
`;
